openshift/helper/__init__.py

```diff
@@ -127,8 +439,220 @@ def __lookup_method(self, operation, namespaced):
+
+        :return: None
+        """
+        logger.debug("arg_spec:")
+        tmp_arg_spec = copy.deepcopy(self.argspec)
+        for key in tmp_arg_spec.keys():
+            if tmp_arg_spec[key].get('no_log'):
+                tmp_arg_spec.pop(key)
+        logger.debug(json.dumps(tmp_arg_spec, indent=4))
+
+    def __transform_properties(self, properties, prefix='', path=[]):
+        """
+        Convert a list of properties to an argument_spec dictionary
+
+        :param properties: List of properties from self.properties_from_model_obj()
+        :param prefix: String to prefix to argument names.
+        :return: dict
+        """
+        args = {}
+        for prop, prop_attributes in properties.items():
+            if prop in ('api_version', 'status', 'kind'):
+                # Don't expose these properties
+                continue
+            elif prop_attributes['immutable']:
+                # Property cannot be set by the user
+                continue
+            elif prop == 'metadata':
+                if 'labels' in dir(prop_attributes['class']):
+                    args['labels'] = {
+                        'type': 'dict',
+                        'property_path': ['metadata', 'labels']
+                    }
+                if 'annotations' in dir(prop_attributes['class']):
+                    args['annotations'] = {
+                        'type': 'dict',
+                        'property_path': ['metadata', 'annotations']
+                    }
+                if 'namespace' in dir(prop_attributes['class']):
+                    args['namespace'] = {
+                        'required': True,
+                        'property_path': ['metadata', 'name']
+                    }
+                if 'name' in dir(prop_attributes['class']):
+                    args['name'] = {
+                        'required': True,
+                        'property_path': ['metadata', 'name']
+                    }
+            elif prop_attributes['class'].__name__ not in ['int', 'str', 'bool', 'list', 'dict']:
+                # Adds nested properties recursively
+
+                # As we traverse deeper into nested properties, we prefix the final primitive property name with the
+                # chain of property names. For example, 'pod_pod_security_context_run_as_user', where 'run_as_user' is
+                # the primitive.
+                #
+                # This may be too hacky, but trying to remove redundant prefixes and generic, non-helpful
+                # prefixes (e.g. Spec, Template).
+                label = prop_attributes['class'].__name__\
+                        .replace(self.api_version.capitalize(), '')\
+                        .replace(BASE_API_VERSION, '')\
+                        .replace(self.base_model_name, '')\
+                        .replace('Unversioned', '')
+                        # .replace('Spec', '')\
+                        # .replace('Template', '')\
```

Comment:
I removed these... but we end up this for a parameter name:

```
spec_pod_template_spec_pod_spec_pod_security_context_se_linux_options_level
```

If I'm going to write a playbook by hand, then I personally would rather see:
```
pod_security_context_se_linux_options_level
```